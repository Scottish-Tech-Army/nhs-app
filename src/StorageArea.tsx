import React from 'react';
import './App.css';
import storageAreaContents from "./data/traumaTower.json";

function StorageArea() {
  return (
    <div>
      <h1>{storageAreaContents.name}</h1>
      <ul>
        {storageAreaContents.boxes.map((box) => (
          <li key={box.boxId} >
            <a href={`/box/${box.boxId}`}>{box.name}</a>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default StorageArea;
