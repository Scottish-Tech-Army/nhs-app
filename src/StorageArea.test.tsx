import { render, screen } from "@testing-library/react";
import StorageArea from "./StorageArea";
import "@testing-library/jest-dom";
import React from "react";

describe("StorageArea", () => {
  it("renders a home page", async () => {
    render(<StorageArea />);

    expect(screen.getByText("Major Trauma Tower")).toBeDefined();

    const links = await screen.findAllByRole("link");
    expect(links.map((link) => link.getAttribute("href"))).toEqual([
      "/box/0",
      "/box/1",
      "/box/2",
      "/box/3",
      "/box/4",
      "/box/5",
    ]);
    expect(links.map((link) => link.textContent)).toEqual([
      "trauma chest drain - box 1",
      "trauma chest drain - box 2",
      "trauma chest drain - box 3",
      "trauma chest drain - box 4",
      "trauma chest drain - box 5",
      "trauma chest drain - box 6",
    ]);
  });

  it("renders correctly", () => {
    const { container } = render(<StorageArea />);

    expect(container).toMatchSnapshot();
  });
});
