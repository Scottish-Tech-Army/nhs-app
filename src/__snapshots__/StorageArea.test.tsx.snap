// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StorageArea renders correctly 1`] = `
<div>
  <div>
    <h1>
      Major Trauma Tower
    </h1>
    <ul>
      <li>
        <a
          href="/box/0"
        >
          trauma chest drain - box 1
        </a>
      </li>
      <li>
        <a
          href="/box/1"
        >
          trauma chest drain - box 2
        </a>
      </li>
      <li>
        <a
          href="/box/2"
        >
          trauma chest drain - box 3
        </a>
      </li>
      <li>
        <a
          href="/box/3"
        >
          trauma chest drain - box 4
        </a>
      </li>
      <li>
        <a
          href="/box/4"
        >
          trauma chest drain - box 5
        </a>
      </li>
      <li>
        <a
          href="/box/5"
        >
          trauma chest drain - box 6
        </a>
      </li>
    </ul>
  </div>
</div>
`;
